sdk/azidentity/environment_credential.go

```diff
@@ -75,9 +86,21 @@ func NewEnvironmentCredential(options *EnvironmentCredentialOptions) (*Environme
 	if clientID == "" {
 		return nil, errors.New("missing environment variable " + azureClientID)
 	}
+	// tenants set by NewDefaultAzureCredential() override the value of AZURE_ADDITIONALLY_ALLOWED_TENANTS
+	additionalTenants := options.additionallyAllowedTenants
+	if len(additionalTenants) == 0 {
```

Comment:
Nit: Won't this always be true, given that the option isn't exported?

Reply:
The option exists so `DefaultAzureCredentialOptions.AdditionallyAllowedTenants` can override the environment variable, preventing weirdness like the option having unexpected or inconsistent effects.